### ISSUE TITLE
grad_fn missing?

### ISSUE THREAD

**Issue by misko** [pytorch/pytorch#16042]
## 🐛 Bug

Hi, Sorry if this is not a bug I tried to post in forums but am getting no response :( 

## To Reproduce

import torch

x = torch.randn(1,1,1,1).requires_grad_(True)

y = torch.nn.Conv2d(1,1,1)(x).mean()**2
conv_grad,  = torch.autograd.grad(y, x, create_graph=True, retain_graph=True)
print("Conv grad",conv_grad)

y = torch.nn.Linear(1,1)(x).mean()**2
linear_grad,  = torch.autograd.grad(y, x, create_graph=True, retain_graph=True)
print("Linear grad",linear_grad)

## Expected behavior

I think both gradient variables should have an associated grad_fn, however only the linear version does. Isn't the math the same for both here, the 1x1 convolution vs the 1x1 linear layer? Or maybe I am missing something :(

**Comment by soumith** [comment]
I think this was flagged as a bug, and a fix is in master. See https://github.com/pytorch/pytorch/issues/15353

It'll be part of the 1.0.1 release this week.
